assets/src/block-editor/helpers/index.js

```diff
@@ -163,68 +130,79 @@ export const addAMPAttributes = ( settings, name ) => {
 };
 
 /**
- * Filters blocks' save function.
+ * Removes `amp-fit-text` related attributes on blocks via block deprecation.
  *
- * @param {Object} element        Element to be saved.
- * @param {string} blockType      Block type.
- * @param {string} blockType.name Block type name.
- * @param {Object} attributes     Attributes.
+ * @param {Object} settings Block settings.
+ * @param {string} name     Block name.
  *
- * @return {Object} Output element.
+ * @return {Object} Modified block settings.
  */
-export const filterBlocksSave = ( element, blockType, attributes ) => { // eslint-disable-line complexity
-	const fitTextProps = {
-		layout: 'fixed-height',
-	};
-
-	if ( 'core/paragraph' === blockType.name && ! attributes.ampFitText ) {
-		const content = getAmpFitTextContent( attributes.content );
-		if ( content !== attributes.content ) {
-			return cloneElement(
-				element,
-				{
-					key: 'new',
-					value: content,
-				},
-			);
-		}
-	} else if ( TEXT_BLOCKS.includes( blockType.name ) && attributes.ampFitText ) {
-		if ( attributes.minFont ) {
-			fitTextProps[ 'min-font-size' ] = attributes.minFont;
-		}
-		if ( attributes.maxFont ) {
-			fitTextProps[ 'max-font-size' ] = attributes.maxFont;
-		}
-		if ( attributes.height ) {
-			fitTextProps.height = attributes.height;
+export const removeAmpFitTextFromBlocks = ( settings, name ) => {
+	if ( TEXT_BLOCKS.includes( name ) ) {
+		if ( ! settings.deprecated ) {
+			settings.deprecated = [];
 		}
 
-		fitTextProps.children = element;
-
-		return <amp-fit-text { ...fitTextProps } />;
-	}
-
-	return element;
-};
-
-/**
- * Returns the inner content of an AMP Fit Text tag.
- *
- * @param {string} content Original content.
- *
- * @return {string} Modified content.
- */
-export const getAmpFitTextContent = ( content ) => {
-	const contentRegex = /<amp-fit-text\b[^>]*>(.*?)<\/amp-fit-text>/;
-	const match = contentRegex.exec( content );
-
-	let newContent = content;
-
-	if ( match && match[ 1 ] ) {
-		newContent = match[ 1 ];
+		settings.deprecated.unshift( {
+			supports: settings.supports,
+			attributes: {
+				...( settings?.attributes || {} ),
```

Comment:
On second thought, the `?` wouldn't be needed. 

```suggestion
				...( settings.attributes || {} ),
```

Reply:
The outcome would be the same, right? But yea, the `?` is redundant.